R/zchunk_LB120.LC_GIS_R_LTgis_Yh_GLU.R

```diff
@@ -0,0 +1,167 @@
+      left_join(SAGE_LT, by = "LT_SAGE") %>%  # includes NAs
+      rename(LT_SAGE_5 = Land_Type) %>%
+      # Drop all rows with missing values (inland bodies of water)
+      na.omit %>%
+      # Reset WDPA classification to "Non-protected" where HYDE classification is cropland, pasture, or urban land
+      mutate(LT_WDPA = if_else(LT_HYDE!="Unmanaged", "Non-protected", "Unmanaged"),
+             Land_Type = LT_SAGE_5,
+             # These multi-tiered classifications will be used for C contents, but for all land cover processing, collapse into GCAM land types
+             Land_Type = if_else(LT_HYDE == "Cropland", "Cropland", Land_Type),
+             Land_Type = if_else(LT_HYDE == "Pasture", "Pasture", Land_Type),
+             Land_Type = if_else(LT_HYDE == "UrbanLand", "UrbanLand", Land_Type),
+             # Area in thousand square kilometers (bm2)
+             Area_bm2 =  value * CONV_HA_BM2) ->
+      L100.Land_type_area_ha
+
+    # LAND COVER FOR LAND ALLOCATION
+    # Aggregate into GCAM regions and land types
+    # Part 1: Land cover by GCAM land category in all model history/base years
+    # Collapse land cover into GCAM regions and aggregate land types
+    L100.Land_type_area_ha %>%
+      group_by(GCAM_region_ID, Land_Type, year, GLU) %>%
+      summarise(Area_bm2 = sum(Area_bm2)) %>%
+      ungroup %>%
+      # Missing values should be set to 0 before interpolation, so that in-between years are interpolated correctly
+      # We dot his because Alan Di Vittorio (see sources above) isn't writing out all possible combinations of
+      # country, GLU, year (of which there are 30), and land use category (of which there are also about 30).
+      # If something isn't written out by the LDS, that is because it is a zero; this step back-fills the zeroes.
+      complete(nesting(GCAM_region_ID, Land_Type, GLU), year, fill = list(Area_bm2 = 0)) %>%
+      # Expand to all combinations with land cover years
+      complete(nesting(GCAM_region_ID, Land_Type, GLU), year = unique(c(year, aglu.LAND_COVER_YEARS))) %>%
+      group_by(GCAM_region_ID, Land_Type, GLU) %>%
+      # Interpolate
+      mutate(Area_bm2 = approx_fun(year, Area_bm2)) %>%
+      filter(year %in% aglu.LAND_COVER_YEARS) %>%
+      arrange(GCAM_region_ID, Land_Type, GLU, year) %>%
+      rename(value = Area_bm2) ->
+      L120.LC_bm2_R_LT_Yh_GLU
+
+    # Subset the land types that are not further modified
+    L120.LC_bm2_R_UrbanLand_Yh_GLU <- filter(L120.LC_bm2_R_LT_Yh_GLU, Land_Type == "UrbanLand")
+    L120.LC_bm2_R_Tundra_Yh_GLU <- filter(L120.LC_bm2_R_LT_Yh_GLU, Land_Type == "Tundra")
+    L120.LC_bm2_R_RckIceDsrt_Yh_GLU <- filter(L120.LC_bm2_R_LT_Yh_GLU, Land_Type == "RockIceDesert")
+
+    # LAND COVER FOR CARBON CONTENT CALCULATION
+    # Compile data for land carbon content calculation on unmanaged lands
+    # Note: not just using the final year, as some land use types may have gone to zero over the historical period.
```

Comment:
I'm not sure I understand this at all. Are we just averaging land over the historical period?